SECP-service/src/main/java/com/visucius/secp/Controllers/User/UserController.java

```diff
@@ -33,4 +44,55 @@ public boolean isUserAnAdmin(String userID) {
 
         return user.get().getLoginRole().equals(LoginRole.ADMIN);
     }
+
+    public Response addDevice(DeviceDTO deviceDTO) {
+        validate(deviceDTO);
+
+        String deviceName = deviceDTO.getDeviceName();
+
+        Optional<User> dbUser = userDAO.find(deviceDTO.getUserID());
+        if (!dbUser.isPresent()) {
+            throw new WebApplicationException(UserErrorMessage.DEVICE_ADD_FAIL_USER_NOT_FOUND, Response.Status.BAD_REQUEST);
+        }
+
+        Device device = deviceDAO.findByDeviceName(deviceName);
+        if(device == null) {
+            device = new Device(deviceName, deviceDTO.getPublicKey());
+            deviceDAO.save(device);
+        }
+
+        List<Device> devicesForUser = deviceDAO.getDevicesForUser(deviceDTO.getUserID());
+        if (devicesForUser == null || !devicesForUser.contains(device)) {
+            //adding the device for the user
+            User user = dbUser.get();
+            user.addDevice(device);
+            userDAO.save(user);
+            LOG.info("New device added for user");
+            return Response.status(Response.Status.CREATED).entity(device.getId()).build();
+        }
+
+        throw new WebApplicationException(UserErrorMessage.DEVICE_ADD_FAIL_DEVICE_EXISTS, Response.Status.BAD_REQUEST);
+    }
+
+    public boolean isDeviceRegisteredForUser(long userID, String deviceName)
+    {
+        List<Device> devicesForUser = deviceDAO.getDevicesForUser(userID);
+        if (devicesForUser == null || devicesForUser.isEmpty()) {
+            return false;
+        }
+
+        Set<String> deviceNamesForUser = devicesForUser.stream().map(Device::getName).collect(Collectors.toSet());
+        return deviceDAO.findByDeviceName(deviceName) != null && deviceNamesForUser.contains(deviceName);
+    }
+
+    private void validate(DeviceDTO deviceDTO) {
+        if (deviceDTO == null) {
+            throw new WebApplicationException(UserErrorMessage.DEVICE_ADD_FAIL_NO_DEVICE_INFO, Response.Status.BAD_REQUEST);
+        }
+        if (StringUtils.isBlank(deviceDTO.getDeviceName())) {
+            throw new WebApplicationException(UserErrorMessage.DEVICE_ADD_FAIL_NO_DEVICE_NAME, Response.Status.BAD_REQUEST);
+        } else if (StringUtils.isBlank(deviceDTO.getPublicKey())) {
```

Comment:
For the public key shouldn't their be other restrictions like the length of the key?